truelearn/learning/_base.py

```diff
@@ -148,92 +153,77 @@ def set_params(self, **args) -> Self:
         for key, value in args.items():
             key, delim, sub_key = key.partition(BaseClassifier.__DEEP_PARAM_DELIMITER)
             if key not in valid_params:
-                raise ValueError(
-                    f"The given parameter {key}"
-                    f" is not in the class {self.__class__.__name__}."
+                raise KeyError(
+                    f"The given parameter {key}" f" is not in the {type(self)}."
                 )
 
             if delim:
                 nested_params[key][sub_key] = value
             else:
-                # ensure that they have the same type
-                if not isinstance(value, type(valid_params[key])):
-                    raise TypeError(
-                        f"The given parameter {key}"
-                        " doesn't have the same type"
-                        " as the original value."
-                        f" Expected {type(valid_params[key])}."
-                        f" Got {type(value)}."
-                    )
                 setattr(self, key, value)
                 valid_params[key] = value
 
         for key, sub_params in nested_params.items():
             valid_params[key].set_params(**sub_params)
 
+        # verify that the new parameters are valid
+        self._validate_params()
+
         return self
 
     @final
-    def _validate_params(self, **kargs) -> None:
+    def _validate_params(self) -> None:
         """Validate types of given arguments in __init__.
 
         Args:
```

Comment:
Remove this as there is no kargs

Reply:
There is no `kargs` in the latest version. Looks like you are reading a previous version?

Reply:
Sorry,im referring to the docstring, args can be removed completely